Content.Server/Animals/Systems/UdderSystem.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Content.Server.Animals.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Content.Server.Chemistry.EntitySystems;
+using Content.Server.Nutrition.Components;
+using Content.Shared.Nutrition.Components;
+
+namespace Content.Server.Animals.Systems
+{
+    /// <summary>
+    ///     Gives ability to living beings with acceptable hunger level to produce reagents.
+    /// </summary>
+    //TODO: Actually it can produce any kind of reagent, so it could be renamed to something more generic...?
+    internal class UdderSystem : EntitySystem
+    {
+        [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+
+        public override void Update(float frameTime)
+        {
+            foreach (var (udder, hunger) in EntityManager.EntityQuery <UdderComponent, HungerComponent>(false))
+            {
```

Comment:
if the animal doesn't have  hungercomponent it should probably still work? the hunger check should only happen if it actually can get hungry in the first place